agent/node.go

```diff
@@ -616,12 +616,14 @@ func (n *Node) runManager(ctx context.Context, securityConfig *ca.SecurityConfig
 			go n.initManagerConnection(ctx, ready)
 
 			go func() {
-				select {
-				case <-ready:
-				case <-ctx.Done():
-				}
-				if ctx.Err() == nil {
-					n.remotes.Observe(api.Peer{NodeID: n.nodeID, Addr: n.config.ListenRemoteAPI}, 5)
+				// this happens only on initial start
+				if ready != nil {
```

Comment:
Actually setting this nil outside goroutine may not be a good idea. We want to `Observe` part to run at least once. So if the first attempt to connect fails we still want to run it on second time.